File: codegen/swagger-ehr/dart/lib/model/attestation.dart
part of swagger.api;

class Attestation {
  
  String type = "ATTESTATION";

  String systemId = null;

  DvDateTime timeCommitted = null;

  DvCodedText changeType = null;

  DvText description = null;

  PartyProxy committer = null;

  DvMultimedia attestedView = null;

  String proof = null;

  List<DvEhrUri> items = [];

  DvText reason = null;

  bool isPending = null;

  Attestation();

  @override
  String toString() {
    return 'Attestation[type=$type, systemId=$systemId, timeCommitted=$timeCommitted, changeType=$changeType, description=$description, committer=$committer, attestedView=$attestedView, proof=$proof, items=$items, reason=$reason, isPending=$isPending, ]';
  }

  Attestation.fromJson(Map<String, dynamic> json) {
    if (json == null) return;
    type = json['_type'];
    systemId = json['system_id'];
    timeCommitted = new DvDateTime.fromJson(json['time_committed']);
    changeType = new DvCodedText.fromJson(json['change_type']);
    description = new DvText.fromJson(json['description']);
    committer = new PartyProxy.fromJson(json['committer']);
    attestedView = new DvMultimedia.fromJson(json['attested_view']);
    proof = json['proof'];
    items = DvEhrUri.listFromJson(json['items']);
    reason = new DvText.fromJson(json['reason']);
    isPending = json['is_pending'];
  }

  Map<String, dynamic> toJson() {
    return {
      '_type': type,
      'system_id': systemId,
      'time_committed': timeCommitted,
      'change_type': changeType,
      'description': description,
      'committer': committer,
      'attested_view': attestedView,
      'proof': proof,
      'items': items,
      'reason': reason,
      'is_pending': isPending
     };
  }

  static List<Attestation> listFromJson(List<dynamic> json) {
    return json == null ? new List<Attestation>() : json.map((value) => new Attestation.fromJson(value)).toList();
  }

  static Map<String, Attestation> mapFromJson(Map<String, Map<String, dynamic>> json) {
    var map = new Map<String, Attestation>();
    if (json != null && json.length > 0) {
      json.forEach((String key, Map<String, dynamic> value) => map[key] = new Attestation.fromJson(value));
    }
    return map;
  }
}

File: codegen/oas-ehr/javascript-closure-angular/API/Client/Attestation.js
goog.provide('API.Client.Attestation');

/**
 * @record
 */
API.Client.Attestation = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.Attestation.prototype.type;

/**
 * @type {!API.Client.DvMultimedia}
 * @export
 */
API.Client.Attestation.prototype.attestedView;

/**
 * @type {!string}
 * @export
 */
API.Client.Attestation.prototype.proof;

/**
 * @type {!Array<!API.Client.DvEhrUri>}
 * @export
 */
API.Client.Attestation.prototype.items;

/**
 * @type {!API.Client.DvText}
 * @export
 */
API.Client.Attestation.prototype.reason;

/**
 * @type {!boolean}
 * @export
 */
API.Client.Attestation.prototype.isPending;

/**
 * @type {!string}
 * @export
 */
API.Client.Attestation.prototype.systemId;

/**
 * @type {!API.Client.DvDateTime}
 * @export
 */
API.Client.Attestation.prototype.timeCommitted;

/**
 * @type {!API.Client.DvCodedText}
 * @export
 */
API.Client.Attestation.prototype.changeType;

/**
 * @type {!API.Client.DvText}
 * @export
 */
API.Client.Attestation.prototype.description;

/**
 * @type {!API.Client.PartyProxy}
 * @export
 */
API.Client.Attestation.prototype.committer;

